builder/oracle/oci/step_get_default_credentials.go

```diff
@@ -0,0 +1,62 @@
+package oci
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	commonhelper "github.com/hashicorp/packer/helper/common"
+	"github.com/hashicorp/packer/helper/communicator"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+type stepGetDefaultCredentials struct {
+	Debug     bool
+	Comm      *communicator.Config
+	BuildName string
+}
+
+func (s *stepGetDefaultCredentials) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+	var (
+		driver = state.Get("driver").(*driverOCI)
+		ui     = state.Get("ui").(packer.Ui)
+		id     = state.Get("instance_id").(string)
+	)
+
+	// Skip if we're not using winrm
+	if s.Comm.Type != "winrm" {
+		log.Printf("[INFO] Not using winrm communicator, skipping get password...")
+		return multistep.ActionContinue
+	}
+
+	// If we already have a password, skip it
+	if s.Comm.WinRMPassword != "" {
+		ui.Say("Skipping waiting for password since WinRM password set...")
+		return multistep.ActionContinue
+	}
+
+	username, password, err := driver.GetInstanceInitialCredentials(id)
```

Comment:
you can bind the context passed in to Run and send it here. It might even be worth inlining that function call if it's not used anywhere else

Reply:
Probably out of scope for this PR as we don't pass the context into any API calls at the moment.

Reply:
#6323